src/app/search/SearchFilter.tsx

```diff
@@ -133,9 +136,34 @@ class SearchFilter extends React.Component<Props, State> {
     }
   );
 
+  private compareMatching = loadingTracker.trackPromise(async () => {
+    const comparableItems = this.getStoresService()
+      .getAllItems()
+      .filter(this.props.searchFilter);
+    CompareService.addMultiItemsToCompare(comparableItems);
+  });
+
+  onQueryChanged(query: string): void {
+    this.props.setSearchQuery(query);
+    loadingTracker.trackPromise(async () => {
+      const filteredItems = this.getStoresService()
+        .getAllItems()
+        .filter(this.props.searchFilter);
+
+      if (!filteredItems.length) {
+        return;
+      }
+      const compareType = CompareService.compareType;
```

Comment:
We shouldn't pull state out of `CompareService` like this. It should be moved to Redux if we want to share it in a bunch of places. 

I think we don't need to worry about using search to add into a compare session though - let's limit this to just opening a new session.